src/components/molecules/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,87 @@
+import { ChangeEvent, FC, useEffect, useState } from 'react'
+import { isValid, parse } from 'date-fns'
+import { Input, InputGroup, InputRightElement, Text, VStack } from '@chakra-ui/react'
+import { CalendarIcon } from '@/components/atoms/CalendarIcon/CalendarIcon'
+
+interface DateInputProp {
+  label: string
+  openDatePicker: () => void
+  value: Date | undefined
+  maxDate?: Date
+  minDate?: Date
+  onChange: (date: Date | undefined) => void
+}
+
+const DateInput: FC<DateInputProp> = (prop) => {
+  const { openDatePicker, label, value, minDate, maxDate, onChange } = prop
+
+  const [textInputValue, setTextInputValue] = useState<string>(``)
+  const [isInvalid, setIsInvalid] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (value) {
+      setTextInputValue(value.toLocaleDateString(`en-sg`))
+    } else {
+      setTextInputValue(``)
+    }
+  }, [value])
+
+  const inputConfig = {
+    height: `28px`,
+    padding: 0,
+    _active: { boxShadow: `none` },
```

Comment:
Why do you use underscores in names?

Reply:
ah, these are how Chakra deals with pseudo classes